### PR TITLE
update button styles for admin

### PR DESCRIPTION
**Pull request by scorpigg**

### CODE REVIEW

File: gf-server-api/DTO/roomsDTO.json [hunk 1]
```diff
@@ -1,81 +1 @@
-[
-  {
```
Comment (tritonJS826) [inline gf-server-api/DTO/roomsDTO.json:2]: Please make your file structure more readable.

File: gf-webapp-admin/src/component/rooms/room/Room.module.scss [hunk 2]
```diff
@@ -14,6 +14,18 @@
...
+    display: flex;
+    align-items: center;
+    width: auto;
+    height: auto;
+    font-size: 1rem;
+    animation: none;
+    font-family: 'Lato Regular', sans-serif, 'Arial';
+    border-radius: 5px;
+    padding: 5px 10px;
+  }
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/Room.module.scss:28]: you already have button in library. Why do u create another style here?

File: gf-webapp-admin/src/component/registration/Registration.module.scss [hunk 3]
```diff
@@ -28,7 +28,7 @@
 }
 
 .registrationBtn {
-  padding: 5px 10px;
+  padding: 5px 20px;
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/registration/Registration.module.scss:31]: you already have button in library. Why do u create another style here?

File: gf-webapp-admin/src/component/rooms/room/promo/Promo.module.scss [hunk 4]
```diff
@@ -55,6 +55,8 @@
   }
 
   button {
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/promo/Promo.module.scss:57]: you already have button in library. Why do u create another style here?

File: gf-webapp-admin/src/component/rooms/room/slider/slides/addSlide/AddSlide.module.scss [hunk 5]
```diff
@@ -25,6 +25,8 @@
   }
 
   button {
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/slider/slides/addSlide/AddSlide.module.scss:27]: you already have button in library. Why do u create another style here?

File: gf-webapp-admin/src/component/rooms/roomSimplified/overlay/Overlay.module.scss [hunk 6]
```diff
@@ -14,6 +14,26 @@
   h3 {
     margin-bottom: 10px;
   }
+
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/roomSimplified/overlay/Overlay.module.scss:17]: you already have button in library. Why do u create another style here?

File: gf-webapp-admin/src/component/sideMenu/SideMenu.module.scss [hunk 7]
```diff
@@ -9,6 +9,11 @@
   align-items: center;
   gap: 30px;
   padding: 20px;
+
+  button {
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/sideMenu/SideMenu.module.scss:13]: you already have button in library. Why do u create another style here?

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by tritonJS826** [comment]
Please specify related issue number in commit

File: gf-webapp-admin/src/component/registration/Registration.module.scss (hunk 3 above)
Comment (Ekaterina1994) [inline gf-webapp-admin/src/component/registration/Registration.module.scss:31]: it is desirable to put in constants paddings, marging etc

File: gf-webapp-admin/src/component/registration/Registration.module.scss [hunk 8]
```diff
@@ -37,6 +37,7 @@
   font-size: 1.1rem;
```
Comment (Ekaterina1994) [inline gf-webapp-admin/src/component/registration/Registration.module.scss:37]: Need to check fonts in all projects. Sometimes its rem, sometimes px, and put it in consts

Comment (tritonJS826) [reply gf-webapp-admin/src/component/registration/Registration.module.scss:37]: we shoul create  global variable for font sizes

File: gf-webapp-admin/src/component/registration/Registration.module.scss (hunk 3 above)
Comment (tritonJS826) [reply gf-webapp-admin/src/component/registration/Registration.module.scss:31]: If it is incapsulated paddings, margin etc. you shouldn't put them in global CSS variables

File: gf-webapp-admin/src/component/rooms/room/Room.module.scss [hunk 9]
```diff
@@ -14,6 +14,18 @@
...
   }
+
+  button {
+    display: flex;
+    align-items: center;
+    width: auto;
+    height: auto;
+    font-size: 1rem;
+    animation: none;
+    font-family: 'Lato Regular', sans-serif, 'Arial';
```
Comment (Ekaterina1994) [inline gf-webapp-admin/src/component/rooms/room/Room.module.scss:25]: Check all files where we don't use familu Lato. If everywhere Lato? then better write it with * in index.scss

File: gf-webapp-admin/src/component/registration/Registration.module.scss (hunk 8 above)
Comment (tritonJS826) [reply gf-webapp-admin/src/component/registration/Registration.module.scss:37]: BTW: please specify global variables in gf-ui-lib (not in each module separate)

File: gf-webapp-admin/src/component/rooms/room/Room.module.scss (hunk 9 above)
Comment (Ekaterina1994) [inline gf-webapp-admin/src/component/rooms/room/Room.module.scss:25]: Check all files where we don't use familu Lato. If everywhere Lato? then better write it with * in index.scss

Comment (tritonJS826) [reply gf-webapp-admin/src/component/rooms/room/Room.module.scss:25]: no, let's put global styles in gf-ui-lib, not in each module separate

File: gf-ui-lib/src/components/Button/Button.tsx [hunk 10]
```diff
@@ -16,6 +16,10 @@ interface ButtonProps {
    * Button type
    */
   type: "submit" | "button";
+  /**
+   * Button size
+   */
+  size?: "content"
```
Comment (tritonJS826) [inline gf-ui-lib/src/components/Button/Button.tsx:22]: * Please specify button behavior when prop size is not defined in comment. 
* it's wrong external interface. Prop size looks like a flag, but it is not.

File: gf-ui-lib/src/components/Button/Button.tsx [hunk 11]
```diff
@@ -25,7 +29,7 @@ export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
   return (
     <button
       type={props.type ?? "button"}
-      className={styles.button}
+      className={`${styles.button} ${props.size === "content" ? styles.content : ""}`}
```
Comment (tritonJS826) [inline gf-ui-lib/src/components/Button/Button.tsx:32]: Ternarian operatorn in template strings in render method in component is too hard to understand. Don't be shy to move out any logic from return. Make return as simple as possible

File: gf-webapp-admin/src/component/rooms/room/RoomItem.tsx [hunk 12]
```diff
@@ -10,6 +10,7 @@ import {Services} from "src/component/rooms/room/services/Services";
 import {Slider} from "src/component/rooms/room/slider/Slider";
 import {Square} from "src/component/rooms/room/square/Square";
 import styles from "src/component/rooms/room/Room.module.scss";
+import {DeleteRoom} from "./deleteRoom/DeleteRoom";
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/RoomItem.tsx:13]: u shoud use absolut path

File: gf-webapp-admin/src/component/rooms/room/apartmentType/ApartmentType.tsx [hunk 13]
```diff
@@ -39,6 +39,7 @@ export const ApartmentsType = (props: ApartmentsTypeProps) => {
           type="button"
           value={isEditFieldDisabled ? "Edit" : "Save"}
           onClick={isEditFieldDisabled ? fieldEditHandler : saveHandler}
+          size="content"
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/apartmentType/ApartmentType.tsx:42]: size = content - its not clear which content (internal or external)

File: gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+interface DeleteRoomProps {
+  id: string;
+}
+
+export const DeleteRoom = (props: DeleteRoomProps) => {
+  const navigate = useNavigate();
+
+  const onDeleteRoom = async () => {
+    await deleteRoom(props.id);
+    navigate("/rooms");
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx:14]: hardcode

File: gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import {Button} from "gf-ui-lib/components/Button/Button";
+import {useNavigate} from "react-router-dom";
+import {deleteRoom} from "src/service/RoomService";
+
+interface DeleteRoomProps {
+  id: string;
+}
+
+export const DeleteRoom = (props: DeleteRoomProps) => {
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx:9]: Dont create new DeleteRoomButton with DeleteRoom name. Do u really need this component?

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by tritonJS826** [comment]
Please ask for each comment. It's hard to understand which comment was completed by our current history.

File: gf-ui-lib/src/components/Button/Button.tsx (hunk 10 above)
Comment (scorpigg) [reply gf-ui-lib/src/components/Button/Button.tsx:22]: Done

File: gf-ui-lib/src/components/Button/Button.tsx (hunk 11 above)
Comment (scorpigg) [reply gf-ui-lib/src/components/Button/Button.tsx:32]: Done

File: gf-webapp-admin/src/component/rooms/room/RoomItem.tsx (hunk 12 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/RoomItem.tsx:13]: deleteRoom has beeen removed

File: gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx (hunk 15 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx:9]: deleteRoom has beeen removed

File: gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx (hunk 14 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/deleteRoom/DeleteRoom.tsx:14]: remade `/rooms` to `MAIN_PAGE_ROUTE` (imported)

File: gf-server-api/DTO/roomsDTO.json (hunk 1 above)
Comment (scorpigg) [reply gf-server-api/DTO/roomsDTO.json]: Done

File: gf-webapp-admin/src/component/rooms/room/Room.module.scss (hunk 2 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/Room.module.scss]: Done

File: gf-webapp-admin/src/component/registration/Registration.module.scss (hunk 3 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/registration/Registration.module.scss:31]: Done

File: gf-webapp-admin/src/component/rooms/room/promo/Promo.module.scss (hunk 4 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/promo/Promo.module.scss]: Done

File: gf-webapp-admin/src/component/rooms/room/slider/slides/addSlide/AddSlide.module.scss (hunk 5 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/slider/slides/addSlide/AddSlide.module.scss]: Done

File: gf-webapp-admin/src/component/registration/Registration.module.scss (hunk 8 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/registration/Registration.module.scss:37]: Done

File: gf-webapp-admin/src/component/rooms/room/Room.module.scss (hunk 9 above)
Comment (scorpigg) [reply gf-webapp-admin/src/component/rooms/room/Room.module.scss]: Done

File: gf-ui-lib/src/components/Button/Button.tsx [hunk 16]
```diff
@@ -16,19 +17,29 @@ interface ButtonProps {
...
+   */
+  size?: "innerContent" | "parentWidth";
 }
 
 /**
  * Button component
  */
 export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
+
+  const buttonSize = props.size ? props.size : "";
```
Comment (tritonJS826) [inline gf-ui-lib/src/components/Button/Button.tsx:32]: Nope) unscalable solution) and size can't be null or underfind due to default props

Comment (tritonJS826) [reply gf-ui-lib/src/components/Button/Button.tsx:32]: Why don't u want to throw prop into return without additional constant?

File: gf-webapp-admin/src/component/rooms/room/RoomItem.tsx [hunk 17]
```diff
@@ -17,6 +21,13 @@ interface RoomProps {
 
 export const RoomItem = (props: RoomProps) => {
 
+  const navigate = useNavigate();
+
+  const deleteWithRedirect = async () => {
+    await deleteRoom(props.room.id);
```
Comment (tritonJS826) [inline gf-webapp-admin/src/component/rooms/room/RoomItem.tsx:27]: Why do u await before navigation?

File: gf-ui-lib/src/components/Button/Button.tsx [hunk 18]
```diff
@@ -16,19 +17,29 @@ interface ButtonProps {
    * Button type
    */
   type: "submit" | "button";
```
Comment (tritonJS826) [inline gf-ui-lib/src/components/Button/Button.tsx:19]: Could u make type prop optional too, please? I suggest to button as default value

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: gf-ui-lib/src/components/Button/Button.tsx (hunk 18 above)
Comment (scorpigg) [reply gf-ui-lib/src/components/Button/Button.tsx:19]: Done

File: gf-ui-lib/src/components/Button/Button.tsx [hunk 19]
```diff
@@ -14,21 +15,32 @@ interface ButtonProps {
...
+   * If size is undefined, then the button will have parentWidth style
+   */
+  size?: "innerContent" | "parentWidth";
 }
 
 /**
  * Button component
  */
 export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
+  const buttonClassNames = `${styles.button} ${props.size && styles[props.size]}`;
```
Comment (tritonJS826) [inline gf-ui-lib/src/components/Button/Button.tsx:32]: Use typescript to define props types component, don't forget about default props (props.size can't be 0, '', null or undefined, don't check it

Comment (scorpigg) [reply gf-ui-lib/src/components/Button/Button.tsx]: add default states to type and size

**Review by tritonJS826 — COMMENTED** [review]
Done

**Pull request merged**